adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapterTest.java

```diff
@@ -856,6 +856,50 @@ public void testMessageLimitExceededForAnEventMessage(final TestContext ctx) {
                 }));
     }
 
+    /**
+     * Verifies that the payload size of an incoming message is calculated based on
+     * the configured minimum message size and reported via metrics.
+     *
+     * @param ctx The test context to use for running asynchronous tests.
+     */
+    @Test
+    public void testPayloadSizeReportedToMetrics(final TestContext ctx) {
+
+        // GIVEN an AMQP adapter with a configured server
+        final VertxBasedAmqpProtocolAdapter adapter = givenAnAmqpAdapter();
+        final DownstreamSender telemetrySender = givenATelemetrySenderForAnyTenant();
+        when(telemetrySender.send(any(Message.class), any()))
+                .thenReturn(Future.succeededFuture(mock(ProtonDelivery.class)));
+
+        // Given a tenant with minimum message size configured as 4 KB
+        final TenantObject tenantConfig = TenantObject.from(TEST_TENANT_ID, true)
+                .setMinimumMessageSize(4 * 1024);
+        when(tenantClient.get(eq(TEST_TENANT_ID), any())).thenReturn(Future.succeededFuture(tenantConfig));
+
+        // WHEN a gateway uploads a message
+        final ProtonDelivery delivery = mock(ProtonDelivery.class);
+        when(delivery.remotelySettled()).thenReturn(true);
+        final String to = ResourceIdentifier.from(TelemetryConstants.TELEMETRY_ENDPOINT, TEST_TENANT_ID, TEST_DEVICE)
+                .toString();
+        final Buffer payload = Buffer.buffer("some payload");
```

Comment:
using something like
```
payload = Buffer.buffer("payload much smaller than 4kb");
```
would improve readability, don't you think?

Reply:
:+1: